src/app/features/notification/notification-rule/notification-condition/notification-condition-editor/notification-condition-editor.component.spec.ts

```diff
@@ -0,0 +1,61 @@
+import { ComponentFixture, TestBed } from "@angular/core/testing";
+import { NotificationConditionEditorComponent } from "./notification-condition-editor.component";
+import { MockedTestingModule } from "app/utils/mocked-testing.module";
+import { MAT_DIALOG_DATA, MatDialogRef } from "@angular/material/dialog";
+
+describe("NotificationConditionEditorComponent", () => {
+  let component: NotificationConditionEditorComponent;
+  let fixture: ComponentFixture<NotificationConditionEditorComponent>;
+  let dialogRef: jasmine.SpyObj<
+    MatDialogRef<NotificationConditionEditorComponent>
+  >;
+
+  beforeEach(async () => {
+    const dialogRefSpy = jasmine.createSpyObj("MatDialogRef", ["close"]);
+
+    await TestBed.configureTestingModule({
+      imports: [
+        NotificationConditionEditorComponent,
+        MockedTestingModule.withState(),
+      ],
+      providers: [
+        { provide: MAT_DIALOG_DATA, useValue: {} },
+        { provide: MatDialogRef, useValue: dialogRefSpy },
+      ],
+    }).compileComponents();
+
+    fixture = TestBed.createComponent(NotificationConditionEditorComponent);
+    component = fixture.componentInstance;
+    dialogRef = TestBed.inject(MatDialogRef) as jasmine.SpyObj<
+      MatDialogRef<NotificationConditionEditorComponent>
+    >;
+    fixture.detectChanges();
+  });
+
+  it("should create", () => {
+    expect(component).toBeTruthy();
+  });
+
+  it("should close dialog when json is changed", () => {
```

Comment:
really? This doesn't seem to happen (I have to click save, not only change the json - which is good this way, I think). Or do I misunderstand this test title?